### PR TITLE
Debian chapter proposal

### PR DESCRIPTION
**Pull request by cpriouzeau**
Signed-off-by: Christophe Priouzeau <christophe.priouzeau@st.com>

### CODE REVIEW

**Comment by cpriouzeau** [comment]
@muvarov, @ilias.apalodimas@linaro.org can you review this Pull Request

File: source/chapter3-debian.rst [hunk 1]
```diff
@@ -5,3 +5,176 @@ Debian
...
+.. code-block:: bash
+
+    mkdir -p script
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/raw-tools/create_raw_from_flashlayout.sh \
+     -O script/create_raw_from_flashlayout.sh
+    chmod +x script/create_raw_from_flashlayout.sh
+
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/files/ledge-stm32mp157c-dk2/FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template -O  FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template
+
+    (./script/create_raw_from_flashlayout.sh FlashLayout_sdcard_armhf_without_boot_firmware.fld  &&
```
Comment (muvarov) [inline source/chapter3-debian.rst]: () are not needed

Comment (muvarov) [reply source/chapter3-debian.rst]: where FlashLayout_sdcard_armhf_without_boot_firmware.fld comes from?

File: source/chapter3-debian.rst [hunk 2]
```diff
@@ -5,3 +5,176 @@ Debian
...
+-----------------
+
+.. code-block:: bash
+
+    mkdir -p script
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/raw-tools/create_raw_from_flashlayout.sh \
+     -O script/create_raw_from_flashlayout.sh
+    chmod +x script/create_raw_from_flashlayout.sh
+
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/files/ledge-stm32mp157c-dk2/FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template -O  FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template
```
Comment (muvarov) [inline source/chapter3-debian.rst]: where FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template is used?

File: source/chapter3-debian.rst [hunk 3]
```diff
@@ -5,3 +5,176 @@ Debian
...
+.. code-block:: bash
+
+    mkdir -p script
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/raw-tools/create_raw_from_flashlayout.sh \
+     -O script/create_raw_from_flashlayout.sh
+    chmod +x script/create_raw_from_flashlayout.sh
+
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/files/aarch64/FlashLayout_sdcard_arm64_without_boot_firmware.fld -O  FlashLayout_sdcard_arm64_without_boot_firmware.fld
+
+    (./script/create_raw_from_flashlayout.sh FlashLayout_sdcard_arm64_without_boot_firmware.fld &&
```
Comment (muvarov) [inline source/chapter3-debian.rst]: () are mot needed

File: source/chapter3-debian.rst [hunk 4]
```diff
@@ -5,3 +5,176 @@ Debian
...
+-----------------
+
+.. code-block:: bash
+
+    mkdir -p script
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/raw-tools/create_raw_from_flashlayout.sh \
+     -O script/create_raw_from_flashlayout.sh
+    chmod +x script/create_raw_from_flashlayout.sh
+
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/files/ledge-stm32mp157c-dk2/FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template -O FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template
```
Comment (muvarov) [inline source/chapter3-debian.rst:221]: shall this template be changed with specifying %%BOOTFS% ?

File: source/chapter3-debian.rst (hunk 2 above)
Comment (cpriouzeau) [reply source/chapter3-debian.rst]: Corrected

File: source/chapter3-debian.rst (hunk 1 above)
Comment (cpriouzeau) [reply source/chapter3-debian.rst]: Corrected

File: source/chapter3-debian.rst (hunk 3 above)
Comment (cpriouzeau) [reply source/chapter3-debian.rst]: corrected

File: source/chapter3-debian.rst (hunk 4 above)
Comment (cpriouzeau) [reply source/chapter3-debian.rst:221]: The script and flashlayout must be aligned for the image filtering

Comment (muvarov) [reply source/chapter3-debian.rst:221]: then it's better to mention about that in the doc.

Comment (cpriouzeau) [reply source/chapter3-debian.rst:221]: yes, but the user cannot see this specificity, it's the script which manage it

Comment (cpriouzeau) [reply source/chapter3-debian.rst:221]: For debian the %%BOOTFS%% are not used, the bootfs are linked to image generated by oe qemu build

Comment (muvarov) [reply source/chapter3-debian.rst:221]: can you please add here what directory structure has to be for this script?

Comment (cpriouzeau) [reply source/chapter3-debian.rst:221]: Sorry, I don't understand what you want ? The script need to have the fld and the debian tarball and for all others files specified on fld which are not present, it dowload it (it the same for qemu)

Comment (muvarov) [reply source/chapter3-debian.rst:221]: ah, ok. In case of download did you found a way to pass linaro authorization?

Comment (cpriouzeau) [reply source/chapter3-debian.rst:221]: No, the credential of linaro are two complicated to be by-passed

**Comment by cpriouzeau** [comment]
Wait before merging, I need to explain deeper the difference between armv7 and armv8

**Comment by muvarov** [comment]
Ok.

**Comment by cpriouzeau** [comment]
I have duplicated the section build steps to be sure the user doesn't mix armv7 and armv8 process due the same name of tarball generated (which are already specified on fld file)

**Review by muvarov — APPROVED** [review]

**Pull request merged**